syntax = "proto3";


service RouteGuide {
    rpc GetTodo(TodoCode) returns (Todo) {}
    rpc ListTodos(Mode) returns (stream Todo) {}
    rpc AddTodo(Todo) returns (stream Todo) {}
    // rpc RecordRoute(stream Point) returns (RouteSummary) {}
    // rpc RouteChat(stream RouteNote) returns (stream RouteNote) {}
}

message Mode {
    enum State {
      COMPLETED = 0;
      YETTODO   = 1;
    }
    State id = 1;
}

message TodoCode {
    int32 id = 1;
}

message Todo {
    TodoCode id = 1;
    string msg = 2;
}
